types/types.go

```diff
@@ -96,7 +96,8 @@ type LayerCompression int
 
 const (
 	// PreserveOriginal indicates the layer must be preserved, ie
-	// no compression or decompression.
+	// no compression or decompression, including already-compressed
+	// layers for which only the compression algorithm is being changed.
```

Comment:
An absolutely non-blocking weak opinion: I kind of think about this case as “a special exception for `BlobInfo.Compression{Operation,Algorithm}`”, not a generic property of `PreserveOriginal`. So maybe document it only there?

OTOH it’s very reasonable to argue that readers of the code are likely to read documentation of the enum value rather than the `CompressionOperation` field.

Completely up to you.

Reply:
Reverting this bit.